mongodb/mongodb-atlas-cli
# CLOUDP-228051: [Atlas CLI] atlas cluster update via config file fails with "HTTP 400 Bad Request (Error code: "INVALID_JSON_ATTRIBUTE")

## Proposed changes



_Jira ticket:_ CLOUDP-228051

This PR updates the command `atlas cluster update <ClusterName> --file <filename.json>` to show a warning message in the 	`os.Stderr` if the configuration file contains a connection string or a cluster name different from the argument.

```bash
./bin/atlas cluster create ClusterM10test --tier M10 --provider AWS --region US_EAST_1
Creating cluster 'ClusterM10test'.

./bin/atlas cluster get ClusterM10test -o json > ~/Downloads/cluster_test.json

./bin/atlas cluster update ClusterM10test --file ~/Downloads/cluster_test.json 2> test.out                                                                            
Updating cluster 'ClusterM10test'.
```


## Bug
`atlas cluster update <ClusterName> --file cluster.json` fails if the cluster config file is generated by `atlas cluster get <ClusterName> > cluster.json`.
## How to reproduce the issue
- Run `atlas cluster create Cluster0 --tier M10 --provider AWS --region US_EAST_1`
- Run `atlas cluster create Cluster1 --tier M10 --provider AWS --region US_EAST_1`
- Run `atlas cluster get Cluster0 -o json > cluster.json`
- Run `atlas cluster update Cluster1 --file cluster.json`

You should get the following error:
```bash
{"detail":"Received JSON for the connectionStrings attribute does not match expected format.","error":400,"errorCode":"INVALID_JSON_ATTRIBUTE","parameters":["connectionStrings"],"reason":"Bad Request"}
Error: https://cloud-dev.mongodb.com/api/atlas/v2/groups/63f4d4a47baeac59406dc131/clusters/Cluster1 PATCH: HTTP 400 Bad Request (Error code: "INVALID_JSON_ATTRIBUTE") Detail: Received JSON for the connectionStrings attribute does not match expected format. Reason: Bad Request. Params: [connectionStrings]
```
## Root Cause:
The endpoint [Modify One Multi-Cloud Cluster from One Project](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Clusters/operation/updateCluster) does not allow to update the cluster connection strings and the cluster Name.





## Checklist



- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)

## Further comments
